utils.py

```diff
@@ -4,6 +4,14 @@
 import csv
 import logging
 import datetime
+import sys
+import traceback
+
+
+# Display exception without re-throwing it.
+def format_last_exception():
+    exc_type, exc_value, exc_traceback = sys.exc_info()
+    return "\n".join(traceback.format_exception(exc_type, exc_value, exc_traceback))
```

Comment:
This is a generic function for outputting the value of an exception, in the same way that the system _would_ display the exception if it wasn't being caught. Helpful for seeing the traceback but not stopping execution.